### PR TITLE
Fsgrid field initialisation

### PR DESCRIPTION
**Pull request by ykempf**
This is the contribution. I recommend that someone else goes thoroughly through it. I can go through carefully tomorrow morning too.

### CODE REVIEW

**Comment by ykempf** [comment]
@tkoskela can you maybe add reviewer suggestions (Markus and me at least, Urs if you like)?

File: MAKE/Makefile.sisu_gcc [hunk 1]
```diff
@@ -85,6 +85,6 @@ INC_EIGEN = -I$(LIBRARY_PREFIX)/eigen/
 INC_DCCRG = -I$(LIBRARY_PREFIX)/dccrg_new_neighbours/
 INC_VECTORCLASS = -I$(LIBRARY_PREFIX)/vectorclass
 #INC_FSGRID = -I$(LIBRARY_PREFIX)/fsgrid
-INC_FSGRID = -I/homeappl/home/koskelat/lib/fsgrid/
```
Comment (galfthan) [inline MAKE/Makefile.sisu_gcc:88]: Should this be the main one, and not the personak copies

**Comment by ykempf** [comment]
I'll also disable the barriers again.

File: Makefile [hunk 2]
```diff
@@ -3,7 +3,7 @@ ARCH = ${VLASIATOR_ARCH}
 #set FP precision to SP (single) or DP (double)
 FP_PRECISION = DP
 #Set floating point precision for distribution function to SPF (single) or DPF (double)
-DISTRIBUTION_FP_PRECISION = DPF
+DISTRIBUTION_FP_PRECISION = SPF
```
Comment (galfthan) [inline Makefile]: Good, let's keep SPF always. It also tests more cases (realf != real)

File: vlasiator.cpp [hunk 3]
```diff
@@ -82,10 +82,10 @@ bool globalflags::balanceLoad = 0;
 ObjectWrapper objectWrapper;
 
 void addTimedBarrier(string name){
-#ifdef NDEBUG
-//let's not do  a barrier
-   return; 
-#endif
+//#ifdef NDEBUG
```
Comment (galfthan) [inline vlasiator.cpp]: Comments should be removed

Comment (ykempf) [reply vlasiator.cpp]: Isn't that this commit now?

File: MAKE/Makefile.sisu_gcc (hunk 1 above)
Comment (ykempf) [reply MAKE/Makefile.sisu_gcc:88]: @tkoskela can oyu comment on the two simultaneous different INC_DCCRG here?

File: vlasiator.cpp (hunk 3 above)
Comment (galfthan) [reply vlasiator.cpp]: ok, fixed now. Good!

**Pull request merged**